istioctl/cmd/tag.go

```diff
@@ -0,0 +1,387 @@
+	cmd := &cobra.Command{
+		Use:   "tag",
+		Short: "Command group used to interact with revision-tags",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.HelpFunc()(cmd, args)
+			if len(args) != 0 {
+				return fmt.Errorf("unknown subcommand %q", args[0])
+			}
+
+			return nil
+		},
+	}
+
+	cmd.AddCommand(tagApplyCommand())
+	cmd.AddCommand(tagListCommand())
+	cmd.AddCommand(tagRemoveCommand())
+
+	return cmd
+}
+
+func tagApplyCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "apply",
+		Short:   "Create or modify revision tags",
+		Example: "istioctl x tag apply prod --revision 1-8-0",
+		Aliases: []string{"create"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("must provide a tag for creation")
+			}
+			if len(args) > 1 {
+				return fmt.Errorf("must provide a single tag for creation")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+
+			return applyTag(context.Background(), client.Kube(), args[0], revision, cmd.OutOrStdout())
+		},
+	}
+
+	cmd.PersistentFlags().BoolVar(&overwrite, "overwrite", false, "whether to overwrite an existing tag")
+	cmd.PersistentFlags().BoolVarP(&skipConfirmation, "skip-confirmation", "y", false, "whether to skip confirmation for tag deletion")
+	cmd.PersistentFlags().StringVarP(&revision, "revision", "r", "", "revision to point tag to")
+	_ = cmd.MarkPersistentFlagRequired("revision")
+
+	return cmd
+}
+
+func tagListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List existing revision tags and their corresponding revisions",
+		Example: "istioctl x tag list",
+		Aliases: []string{"show"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return fmt.Errorf("tag list command does not accept arguments")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+			return listTags(context.Background(), client.Kube(), cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+func tagRemoveCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "remove",
+		Short:   "Remove an existing revision tag",
+		Example: "istioctl x tag remove prod",
+		Aliases: []string{"delete"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("must provide a tag for removal")
+			}
+			if len(args) > 1 {
+				return fmt.Errorf("must provide a single tag for removal")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+
+			return removeTag(context.Background(), client.Kube(), args[0], skipConfirmation, cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+// applyTag creates or modifies a revision tag
```

Comment:
Function comments should be full sentences: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences